artiq/frontend/artiq_client.py

```diff
@@ -25,7 +25,7 @@ def clear_screen():
         sys.stdout.write("\x1b[2J\x1b[H")
 
 
-def get_argparser():
+def get_argparser() -> argparse.ArgumentParser:
```

Comment:
Why the type annotation? Python ignores them.

Reply:
yes, but I don't when I read the code. And it helps IDE's autocomplete better. And if you ever use `mypy` or `pylint` or some other type checker.

Reply:
They are not used anywhere else in ARTIQ, and are generally not common practice when developing Python programs.

Reply:
See my thoughts in https://github.com/m-labs/artiq/pull/1188#discussion_r230823647